homeassistant/components/daikin/config_flow.py

```diff
@@ -40,8 +40,8 @@ def schema(self):
 
     async def _create_entry(self, host, mac, key=None, uuid=None, password=None):
         """Register new entry."""
-        # Check if mac already is registered
-        await self.async_set_unique_id(mac)
+        if not self.unique_id:
+            await self.async_set_unique_id(mac)
```

Comment:
We're not supposed to use different kinds of unique_id for different config flow sources. If we do that it won't work as expected. A zeroconf discovered flow should not be able to compete with a manual flow for the same device.

Reply:
Yes I agree, but `mac` is not available in `async_step_zeroconf` (https://github.com/home-assistant/core/pull/36571/files/32d265d8d5e5c90b0206558c379762d522fba912#diff-9ba0b5cb03cb9763543902445049c666R131). This ensures that the dialog doesn't show up for devices discovered (and added) by zeroconf added devices.

To be honest, I think that the zeroconf support should be dropped if this quite ugly fix doesn't go into 0.111 (to not introduce new issues). I don't have any Daikin devices with zeroconf support so it's been a walk in the dark with this implemenatation.

Reply:
Yeah, I think we can add this as a temporary fix. Then for the the next release remove zeroconf support if there's no solution.

Reply:
> remove zeroconf support if there's no solution

I think that is what we have to do..